### DISCUSSION TITLE
some improvements

### DISCUSSION THREAD

**Discussion by octane321** [Sleeper85/esphome-jk-bms-can#21] [Ideas]
works great so far - i have some improvements requests

- Use BMS max charge/discharge current (-5%) [tested and working]
- recover globals (charge/float voltage, absorbtion time etc) after restart (or is there any disadvantage like wearing?) [tested and working]
- add ability to change CAN Protokoll and Name dynamic (for testing inverter communication)
- use celltemperatures in a relevant way (if temp <20°C, use min temp of cells, if >20 and <40°C use average temp, if >40°C use max temp)
- seperate charge logic into new .yaml and include it as a package

what are your thoughts about that?

THX

**Comment by Sleeper85** [comment]
Hi,

There are some interesting points like that of temperature, I'm about to add an "Alarms Logic" section.

I have a lot of requests for improvement and some will not suit everyone.
I think I should collect all the requests and hold a vote for each question.

I note your requests.

**Comment by virus100b** [comment]
Hello,

@ Sleeper85 Congratulation for the nice project you have done 
I have moddifed liitle your code adding the option of modifying discharge current from slider. I am also planning to implement limiting discharging current to 10% from normal discharging current when battery cappacity drop below 15%. I know some BMS has this function. 

What is you oppinion about this?

**Comment by Sleeper85** [comment]
I have already added an "Alarms Logic" section to address the [@arzaman](https://diysolarforum.com/members/69458/) and [@hxx](https://diysolarforum.com/members/21118/) balancing problem. I'm going to add temperature control to it. Thus, this section will aim to detect problems before the BMS cut-off which will prevent it from working its MOSFETs to cut a lot of amps.

Here's what will be part of version 1.16.5 :

- High temperature control (45°C) => stop charging and discharging
- Low temperature control (0°C) => stop charging
- Cell max voltage control (Bulk V. / 16) => reduce the charging amps to 5A and start a balancing phase
- Cell min voltage control (min_discharge_v /16) ? or bms_cell_undervoltage_protection + 0.x ? => stop discharging

This could be based on information from the BMS or values in the script.
If the BMS is poorly programmed then it will not be useful.

I can take advantage of this to add some of your requests such as the temperature sent (min or max) or a new slider for the discharge current.

But some of your requests are special and may not suit everyone.

FYI: I'm preparing for a long 4 month trip, version 1.16.5 will be the last one I can test before leaving.

**Comment by octane321** [comment]
Didnt read all about the problems from arzaman and hxx but could imagine what problems they have and the solution sounds good.
I would appreciate if the new Alarm logic will be based on the bms values - a poorly configured BMS is always a problem, but if you want to implement some values to set it would be not a bad idea.

i also understand if some requests will not suite to everyone - i´ll try to implement them for personal needs then if i´m able to.
Looking forward to 1.16.5 ;-)

I wish you a wonderful time on the trip

**Comment by virus100b** [comment]
High temperature control (45°C) => stop charging and discharging, i believe is temperature of the cells measured by the two external probes, as the mosfets are not "suffering" by working at 45°C.
Regarding my plan with 10% of the current  limit function i can understand that it my not be suitable for others and i can tweak my code for that. For now i will wait for your final reales as the new feature are nice to have.

Some question:
1. Why did you name the MOS temperature "tube temperature"? Was a bit confusing at the begining as inside BMS they are called MOS.
2.   Discharging low temperature and Discharging low temperature recovery i cannot find in my BMS settings thus are present in the ESP home. This temperature may be hidden?

![Discharging Temp](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/c01737fd-6d9b-4a4f-9da5-5f9580d5a2e0)

P.S. The inverter i have is a  Deye sun 12k-sg04lp3-eu. I have seen that is already in the confirmed working inverters, but i can reconfirms that is working fine ESP32 and TJA1050. 

For power the ESP i have use 12V that inverter provide on CN2 pins 7, 8 and a stepdown converter. Maybe for Deye owners my be useful info spare using a 48-5V Power supply.

**Reply by lucize** [reply]
I'm using it with an mcp2515 can adapter connected at 3.3v. for me tja didn't work, but maybe it was faulty, maybe I'll test again sometime. esp32 is externally 5v powered

**Comment by octane321** [comment]
@virus100b Good point on the High temperature control
the 45°C should be triggered by the cell votages (external probes)
But there is also a MOS OTP and OTPR which could also trigger an alarm.
Would be great to limit the current by x% in both cases instead of shutting (dis)charging completely down

But i don´t get the point of limiting the current if SoC<15%
This will only push the inverter in a range where the efficiency is really bad due to the constant selfcomsumption. And it does not harm the cells or BMS to discharge at the normal rate - so why would you do that?


> 1. Why did you name the MOS temperature "tube temperature"? Was a bit confusing at the begining as inside BMS they are called MOS.

The transistors which control the charging and discharging function of the BMS are commonly called "MOS Tubes"

> 2\. Discharging low temperature and Discharging low temperature recovery i cannot find in my BMS settings thus are present in the ESP home. This temperature may be hidden?

these settings are only available on the JK Inverter BMS i think - but i might be wrong...there are so many different Hardware and Software Versions out there

**Comment by Sleeper85** [comment]
@octane321 The temperature sent with ID 0x356 has an informative role. There are other parts in the code where the minimum or maximum temperature is communicated or used. So I opted to simply send the average temperature.

@virus100b I added the "Discharging current max" slider.

Compiled at home but haven't had time to do all the tests yet.

V1.16.5 Sleeper85 : Add Preventive Alarms Logic with Balancing function, CAN ID 0x356: send average temperature of T1/T2, new "Discharging current max" slider

**Comment by virus100b** [comment]
Hello, 

@Sleeper85 Thanks for new version, if i have time i will try it today. Enjoy and good luck in your trip!


@octane321  "But i don´t get the point of limiting the current if SoC<15%.

I think the idea for limiting the current close to end of the capacity is because on high loads the cell voltage drops and you may switch off the discarding before reaching desired discharge capacity. 

On the charging the current is limiting itself because of constant voltage charging and  i see no point limiting unless poorly setup BMS command the inverter a higher voltage (or a fault in inverter) and then the current is obviously high.

**Comment by Sleeper85** [comment]
Here is the latest version (beta) with the changelog.

If you have time to test it, I would like to have your feedback before publishing it on GitHub.

Thanks

[V1.16.6](https://diysolarforum.com/threads/jk-bms-can-bus-comms-now-possible-for-inverters-that-support-goodwe-and-pylontech-batteries.48963/page-31#post-1002153)

**Comment by virus100b** [comment]
I see that balancing current entity is missing from the wired version. BMS is not exposing that parameter on 485 port?

**Comment by Sleeper85** [comment]
Exactly, there are differences between the information available from BLE
and Wire.

**Comment by virus100b** [comment]
Has anyone tested the code with ESP32 C3 board? I have readed that on S2 CAN is not working. How about C3?

**Comment by shvmm** [comment]
> Has anyone tested the code with ESP32 C3 board? I have readed that on S2 CAN is not working. How about C3?

S2 will not work. The ESPRESSIF docs explicitly state that it lacks support for TWAI (CAN).

https://www.espressif.com/en/products/socs

**Comment by Sleeper85** [comment]
Atom ESP32-S3 work.

**Comment by octane321** [comment]
> Here is the latest version (beta) with the changelog.
> 
> If you have time to test it, I would like to have your feedback before publishing it on GitHub.
> 
> Thanks
> 
> [V1.16.6](https://diysolarforum.com/threads/jk-bms-can-bus-comms-now-possible-for-inverters-that-support-goodwe-and-pylontech-batteries.48963/page-31#post-1002153)

Hi @Sleeper85 ,
hope you are doing well!

Just spent the last couple of days to build an add-on with addressable lights to show the status and SoC - but more on that if its completely done.
I just wanted to test it with the new 1.16.6 beta but saw you removed the link on DIY Forum :-(

Have you already left? I wish you a great time on your trip

**Comment by Sleeper85** [comment]
Hi,
I'm leaving next week.
The latest version 1.17.2 will be available soon. And if you want I can share it with you privately if you want to test it ;-)

**Comment by virus100b** [comment]
Hi @Sleeper85 ,

Happy to test, even i am 2000km away of my home i have remote access of my HA so i can flash it.
What are the goodies for the version 1.17.2 ?

In the future i want to build another battery pack  as i will  like more storage capacity  (currently i have a 280Ah battery) and i will like to monitor that battery too. 
I will try to modify the code for that at some point. 
I believe the simplest way is use the same board using the one free UART for connecting to the second BMS and send to inverter the sum of SOC, the rest of the settings will be the same. HA entities can be published separately(e.g. SK#01... and JK#02...) Right?

**Comment by Sleeper85** [comment]
We still need to fix one more thing before releasing the latest version. It will be soon.

@syssi offers multi-devices UART and BLE YAML. I monitor 2 JK-BMS (BLE) with a single board (without CAN) so yes it's possible. Concerning UART, it's maximum 3 connections with ESP32.

- V1.17.2 MrPablo   : Added function "Auto Charge/Discharge Current Control" to avoid OVP/UVP alarms
- V1.17.1 Sleeper85 : New Cut-Off Current/Voltage Charging Logic for LFP with the participation of @shvmm
- V1.16.6 Sleeper85 : Selectable CAN settings + Adding inverter_offset_v + Improved CAN ID 0x355, sending 100% only at the end of the absorption phase, adding bytes [04:05] and [06:07] + Automatic calculation of the number of battery modules + Save and Restore slider values
- V1.16.5 Sleeper85 : Add Preventive Alarms Logic, CAN ID 0x356: send average temperature of T1/T2, new "Discharging current max" and "Balancing current max" slider

**Comment by virus100b** [comment]
@Sleeper85 

> 1.  I monitor 2 JK-BMS (BLE) with a single board (without CAN) so yes it's possible. Concerning UART, it's maximum 3 connections with ESP32.  

This option is good but i want to send sum of  both batteries SOC to the Inverter as i am using "Time of use" table to stop discharging the battery.

I have modified the name for cells 1-to 9 in 01 to 09 as they where not showing in the right order on entities list. 1, 10...so on, you may modify that if you like in the last update.
![cells](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/af43e91b-c328-48d1-9611-e25fccd7166a)

**Comment by octane321** [comment]
> Hi, I'm leaving next week. The latest version 1.17.2 will be available soon. And if you want I can share it with you privately if you want to test it ;-)

Good morning, would be very nice to get the actual version - so i can implement my things.
I don´t know if you can send it on github somehow - but i´m also on diy forum (sent you a PM)

**Comment by octane321** [comment]
> @Sleeper85
> 
> > 1. I monitor 2 JK-BMS (BLE) with a single board (without CAN) so yes it's possible. Concerning UART, it's maximum 3 connections with ESP32.
> 
> This option is good but i want to send sum of both batteries SOC to the Inverter as i am using "Time of use" table to stop discharging the battery.
> 
> I have modified the name for cells 1-to 9 in 01 to 09 as they where not showing in the right order on entities list. 1, 10...so on, you may modify that if you like in the last update. 

i renamed them too for the correct order (the little Mr.Monk in me strictly demanded that)

I have 3 BMS (with 280Ah cells) and i tried to get two BMS running on a normal ESP32 with BLE, but the BLE Overhead is just massive so quickly running out of heap when using a addressable light strip too.
I deceided to switch to an ESP32-S3 N16R8 now with 8MB of PSRAM. Even if the PSRAM is completely empty, i got much more free heap now. Maybe i´ll give it another try connecting all 3 BMSs together with the CAN communication.
atm i´m running 4 devices - but it would be much safer if only one device deals with everything.

**Comment by Sleeper85** [comment]
@virus100b @octane321 

Ok, cell number renumbering will be part of the next release.

The new YAMLs will be published early next week.

**Comment by Sleeper85** [comment]
[Version 1.17.3 is released](https://github.com/Sleeper85/esphome-jk-bms-can/releases/tag/V1.17.3)

**Comment by virus100b** [comment]
Hello,  
I  have flashed last night with last version and i see that balancing is not working. 
Cell delta is over 0.005v, cells are over staring balance voltage set at 3.45V... Yesterday was working fine with previous version. 

What could be the problem?
![Screenshot_20240224_133432_Home Assistant](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/c77f09a5-c1be-4139-9f1c-26d3f5add870)

**Comment by virus100b** [comment]
Now is working. I just reduce the bulk voltage to force discharge and then raised to 55.2 and is working.  I think is not related to esp. May be BMS was blocked some how. Strange.

![Screenshot_20240224_135607_Chrome](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/12276c7f-d930-43cc-ad66-4dce7c7ae217)

**Comment by Sleeper85** [comment]
Balancing status is purely informative.
It will be displayed when your BMS is equalizing the cells.
We tested it for a week and it worked well for us.

The most important feature for those with an unbalanced battery is the **Auto Charge/Discharge Current Control** feature.

See @MrPabloUK's comment on the subject below:

https://diysolarforum.com/threads/jk-bms-can-with-new-cut-off-charging-logic-open-source.79325/

**Comment by octane321** [comment]
installed it too and working perfectly fine.
Only a little issue with the balancing state:
the balancer is turning off just for a couple of seconds before the balancing is realy finished (at least on one of my BMSs), then mode changes to float and in the meanwhile, the balancer starts balancing again until the total voltage drops too low caused by float mode.
I think a 5-15 min delay would help to prevent this

what do you think?

**Reply by virus100b** [reply]
I understand that balancing notification  is just for info. 
Balancing state switch is actually reporting if balancer inside BMS is working or not and that is was OFF while supposed to be ON as all condition were meet (delta cell voltage over 0.005V,  balance starting voltage over 3.45V) 
I hope that is was a miscommunication in data flow and delta cell voltage was shown over 0.005V and actually was below triggering balance to OFF cause i don't think anything else causing balance to stay OFF.  At the end after i have lower and then rise the bulk voltage,  balancing started and it was working fine. I will see today how is going. 

@Sleeper85  Thanks again for your work and good luck in your trip!

**Comment by Sleeper85** [comment]
For the charge to complete, the 3 conditions below must be met.
* Current <= cutoff current
* Voltage >= cutoff voltage
* The cells are balanced (the balancer is not working).

The Float status will not be replaced by Balancing.

**Reply by virus100b** [reply]
Yesterday after i have cycled bulk voltage up and down I was able to see EOC, and then Float, so all good.

**Reply by octane321** [reply]
I think you are mixing up things - my "issue" is not the same as the one from Virus100b
It is clear, that the bms decides to balance or not. On your case, there must be something with the bms communication.

In my case, the charging state of the esp switches too soon from bulk-voltage to float voltage, cause the bms told the esp that balancing is complete for a very short time. (maybe there was a communication issue too, but i dont think so).
i think the deviation was just a really short amount of time smaller than the differencial trigger voltage and the bms decides the balaning is complete. after balancing turns off, the deviation rised a bit and the bms starts abalncing again.
Meanwhile, for the esp all conditions where met to switch from bulk to float voltage and the charging voltage dropes slowly, what causes the bms to stop balancing caus the cell voltage requirenments where not met anymore.

i´m talking about a deviation of only 5-7 mV - so the cells where already balanced nearly complete.
My idea is, the esp should keeps the bulk voltage for a couple of minutes after all conditions where met to prevent this behavour. Its likely some kind of debouncing.

Hope you understand what i mean

**Reply by Sleeper85** [reply]
@octane321

I completely understand what you are explaining. This is how the new logic is programmed, there is no longer an absorption timer. If you want to extend the end of charge a little you can reduce the Absorption offset V. This will have the effect of increasing the cutoff_v.

**Reply by octane321** [reply]
My Absorption offset V. is set to 0.1V
I did not quiet understood what the Absorption offset V. means - how does it work?

**Reply by Sleeper85** [reply]
In the new version it is mainly used to calculate the cutoff voltage. Go see in the README I published the formulas to calculate cutoff current and voltage.

If you reduce it to 0.05V the end of charge will last longer.

**Reply by octane321** [reply]
just found the thred on diy forum - i´ll give it a try on 0.05 tomorrow 
thx

**Comment by virus100b** [comment]
@Sleeper85  Hello! I hope you are doing well on your trip.
I just found an issue and i don't know if is related to my BMS communication or is a firmware bug. Currently battery is discharging and  ESP is reporting as Balancing ON, Charging status as BULK. If logged on bluetooth directly to BMS,  balancing status is OFF ().  
While battery is charging and the one cell voltage is triggering balance to ON, Charging status is changing to Balancing (Connected to Bluetooth i can see that Balancing is ON), so the process is working OK just information on balancing is not updating to OFF once the Balancing is done. Sometime is turning to OFF but mostly stay ON
Any thoughts about this? 

![JK Balance](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/d38b24a2-3252-44b1-8df8-c43d7e38efae)

**Reply by octane321** [reply]
Hi @virus100b 
would you be so kind and open a issue with your complete config please?

**Comment by virus100b** [comment]
Two days ago i have switched Off BMS for rearranging position, i disconnect the ESP from GPS port but i did not switched off. After i connect ESP and started BMS all worked fine.  Next day i have observed that Daily charge and discharged energy was not reseted. I was expecting to be reseted today but unfortunately still going up. Today have rebooted  ESP and  hopefully this will solve the issue. 
I am wandering what is the reason for not resetting? 

![Daily charge](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/703dfbe0-fc3f-4431-a8ce-229597c4c617)


[
![Daily C D](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/5ec05e5c-e1f3-426d-a0a2-b8ec7c1ad661)
](url)

**Reply by syssi** [reply]
What do you use a time source? If no time is available the component is unable to reset the counters at midnight.

**Reply by virus100b** [reply]
I don't konw what time source i use. I think time source of HomeAssistant.. It was working fine until i have stopped BMS.

**Reply by syssi** [reply]
Take a look at your YAML. There should be a `time` section:

```
time:
  - platform: sntp
```

**Reply by virus100b** [reply]
![Screenshot_20240526_144203_Home Assistant](https://github.com/Sleeper85/esphome-jk-bms-can/assets/64686560/b3438fd4-d8cf-42f8-9223-12d13175a792)

Yes.  I have that time service.

**Reply by Sleeper85** [reply]
Hi Syssi, 

Glad to see you here, I would like to take this opportunity to thank you for all the BMS components you have developed.

@virus100b In the new version which is in the dev branch, I replaced the sntp platform with homeassistant. 

```YAML 
time:
  - platform: sntp
```
By

```YAML 
time:
  - platform: homeassistant
```

I haven't tested it myself but some people have had problems with sntp.

**Comment by virus100b** [comment]
@Sleeper85  Thank you for info.
@syssi  I must say that initially i have used your code and worked fine, but Sleeper85 code is more addapted to my needs. I am using this opportunity to Thank you for your work.

All the best! 
Bogdan.

**Comment by virus100b** [comment]
The board i have used for the last months with no issues on my 12Kw Deye,  i give it to a friend who has a similar inverter like mine V Tac 12kw.
After the batteries are charged and the current is dropping,  the voltage battery is going up to 57V and then dropping down to around 55V and again is going up  to 57 and so on... wile the voltage is up to 57V sometime cells are reaching 3.6V and BMS is reporting alarm. The Float state cannot be reached as the voltage is fluctuating. 

![WhatsApp Image 2024-07-31 at 15 20 35](https://github.com/user-attachments/assets/4fcb5ad6-41f5-4d20-9d43-bd6b59cbb006)
On battery page Charging Voltage sent by Can is 55.3V.  This behavior is triggering when at mid day after battery is reaching full charge and the solar production is high. If some clouds appear and the production is decreasing then this Voltage oscillations stops and everything is working fine.  If the battery mode is changed from LIthium to % this abnormal fluctuating battery voltage is  disappearing and the battery voltage stays constant at the 55.3 as set in menu for the absorption.  
The ESP is  powered from usb charger with 5.1V directly on Vin Pin and to TJA1050 Can module. I have measured termination resistor from CAN and it reed 120Ohm on module and 60Ohm when connected to Invertor. Automatic Cand Voltage, Current is disabled.. I cannot explain what is the problem as the same board on my inverter worked perfectly over 5 months. Any advise what to do?

**Reply by pentierazvanionut** [reply]
Hello everyone
I am the owner. I am attaching the 24 hour battery graph.
even in float I have these fluctuations that last a few minutes.
![IMG_9937](https://github.com/user-attachments/assets/d2c8abc6-e39c-4ddd-a0b6-d78127862987)

**Comment by Sleeper85** [comment]
@virus100b @pentierazvanionut 

What is the version of JK-BMS-CAN or YamBMS?

Can you send the `Requested Charge Voltage` graph?

![image](https://github.com/user-attachments/assets/724932ab-4c9a-43cf-8518-4917845598d2)

**Reply by pentierazvanionut** [reply]
![IMG_9940](https://github.com/user-attachments/assets/cbfcca58-74cc-4c84-965e-f92f34f33a2e)
![IMG_9941](https://github.com/user-attachments/assets/93523a7b-eede-4719-94ee-fd94d09ad1a4)

**Reply by pentierazvanionut** [reply]
I tested the transition from the inverter to the percentage on the battery, in this way it maintains the battery voltage perfectly, only on bms Lithium batt it manifests itself.

**Comment by Sleeper85** [comment]
@virus100b @pentierazvanionut 

I also notice that you are using the `Automatic Charge Voltage` feature of @MrPabloUK  which is still in `V1` in JK-BMS-CAN `1.17.5`.
This feature has been completely rewritten by @MrPabloUK and is in version `2.4.1` in YamBMS `1.4.2`.

So you can also test the behavior of the current ESP32 firmware by disabling the `Auto CVL` feature.

You can also install [YamBMS 1.4.2](https://github.com/Sleeper85/esphome-jk-bms-can/tree/development) which contains many improvements over JK-BMS-CAN 1.17.5, also works with a single BMS.

![image](https://github.com/user-attachments/assets/5efeeb5a-4a3b-41a1-ae86-8e5d59d3998b)

**Comment by Sleeper85** [comment]
@pentierazvanionut 

The graph you just sent does not show what is happening on August 2nd from 11:00 to 14:00.

**Reply by pentierazvanionut** [reply]
![jhhhh](https://github.com/user-attachments/assets/1745221a-4e0a-4796-89c4-8eb620a21836)

**Comment by Sleeper85** [comment]
@virus100b Do you remember what JK-BMS-CAN version was flashed on this ESP32 ?

We can see that the max charge voltage requested is 55.3V which rules out a problem coming from the ESP32.

[This reminds me of this problem with Deye.](https://diysolarforum.com/threads/sol-ark-15k-battery-voltage-regulation-issues.79877/)

@pentierazvanionut What is the exact model, HMI and MAIN version of your Deye ?
Is this the same HMI/MAIN version as @virus100b ?

What do you do when your solar production exceeds the load and the battery is full ?
Do you export this energy to GRID ?

How is your Deye configured ?
Work mode ?
Time Of Use ? (yes/no)
Energy pattern ? (load/battery first)

In % mode what are your Float/Absorption/Equalization charge voltages ?

I know that the Deye inverter does not always respect the requested charging voltage depending on how it is configured, connected or not to the GRID network, exporting or not excess energy.

**Reply by pentierazvanionut** [reply]
The hmi/main version is not the same
my version is Hmi 1001-C043 and main 2006-1144-1807
@virus100b is Hmi 1001-C044 and main 2006-1144-1807
The energy is delivered to the network
It is configured in load first, zero export ct and solar sale
in mode % absorption 55.2 float 54 equalization 55.2

**Reply by virus100b** [reply]
@Sleeper85  Do you remember what JK-BMS-CAN version was flashed on this ESP32 ?  Yes. is 1.17.5. I have modified to send SOC 100%  instead of 99 immediately when 100% is reported by BMS and with this modification we observed less fluctuation, but still happening.

We can see that the max charge voltage requested is 55.3V which rules out a problem coming from the ESP32. I am also thinking that this behavior is coming from FW in the Inverter.

In % mode what are your Float/Absorption/Equalization charge voltages ? 
Float = 54.0, Absorption and Equalisation 55.2 same like in ESP code.
Inverter is set to Load First and the excess is selled to grid.
Time of Use is activated and Grid and Gen charge off, Batt  % between 20 - 45%
My version HMI is 1001-C044
Main 2006-1144-1087
I have Daye and @pentierazvanionut has V TAC.

**Comment by virus100b** [comment]
Good news! 
We find the problem causing the abnormal charge  voltage fluctuations for @pentierazvanionut  V TAC inverter.

Issue was Max Solar Power  being set to 11.000W when the total PV power was little over 11000W.  Value was set to 12.000 and the issue magically dissappear. Fo me this setting don't make sende since the production was not over 11kW when the fluctuations occured. But i want to share our experiences and hope to help others if they have similar issues.

 The firmware was updated by Deye support to last version but this  did not solve the fluctuation issue until we have notice and change Max solar power from 11000 to 12000. Later on we have change this value to 13500 just for safety.  Inverter is working fine for 2 days now. After bulk charge finish, charge voltage is going to float making a flat line of 53.6V.

**Comment by Sleeper85** [comment]
@virus100b 

Cool thanks for sharing the solution with others, I'll keep that in mind.
